challenge-react/src/App.js

```diff
@@ -1,111 +1,197 @@
+    margin-left: 15px;
+    color: white;
+    font-weight: bold;
+    float: right;
+    font-size: 22px;
+    line-height: 20px;
+    cursor: pointer;
+    transition: 0.3s;
+    &:hover {
+      color: black;
     }
+  }
+`;
+const App = () => {
+  const [charities, setCharities] = useState([]);
+  const [selectedAmount, setSelectedAmount] = useState(10);
+  const [showDonate, setShowDonate] = useState(0);
+  const [dataPayments, setDataPayments] = useState([]);
 
-    render() {
-      const self = this;
-      const cards = this.state.charities.map(function (item, i) {
-        const payments = [10, 20, 50, 100, 500].map((amount, j) => (
-          <label key={j}>
-            <input
-              type="radio"
-              name="payment"
-              onClick={function () {
-                self.setState({ selectedAmount: amount });
-              }}
-            />
-            {amount}
-          </label>
-        ));
+  const donate = useSelector((state) => state.donate);
+  const message = useSelector((state) => state.message);
 
-        return (
-          <Card key={i}>
-            <p>{item.name}</p>
-            {payments}
-            <button
-              onClick={handlePay.call(
-                self,
-                item.id,
-                self.state.selectedAmount,
-                item.currency
-              )}
-            >
-              Pay
-            </button>
-          </Card>
-        );
-      });
+  const dispatch = useDispatch();
+
+  const clearAlert = () => {
+    dispatch({
+      type: "UPDATE_MESSAGE",
+      message: "",
+    });
+    clearTimeout(triggerClearMsg);
+  };
 
-      const style = {
-        color: 'red',
-        margin: '1em 0',
-        fontWeight: 'bold',
-        fontSize: '16px',
-        textAlign: 'center',
-      };
+  useEffect(() => {
+    fetch("http://localhost:3001/charities")
+      .then((resp) => {
+        return resp.json();
+      })
+      .then((data) => {
+        setCharities(data);
+      });
 
-      const donate = this.props.donate;
-      const message = this.props.message;
+    fetch("http://localhost:3001/payments")
+      .then((resp) => {
+        return resp.json();
+      })
+      .then((data) => {
+        setDataPayments(data);
+        dispatch({
+          type: "UPDATE_TOTAL_DONATE",
+          amount: summaryDonations(data.map((item) => item.amount)),
+        });
+      });
+  }, []);
 
-      return (
-        <div>
-          <h1>Tamboon React</h1>
-          <p>All donations: {donate}</p>
-          <p style={style}>{message}</p>
-          {cards}
-        </div>
-      );
-    }
-  }
-);
+  const handlePay = (id, amount, currency) => {
+    clearAlert();
 
-/**
- * Handle pay button
- * 
- * @param {*} The charities Id
- * @param {*} amount The amount was selected
- * @param {*} currency The currency
- * 
- * @example
- * fetch('http://localhost:3001/payments', {
-      method: 'POST',
+    fetch("http://localhost:3001/payments", {
```

Comment:
I expect we should use promise/async here for clean code. This is like a callback hell which is hard to read

Reply:
No error handle here?